Modules/TPC/src/PadCalibrationCheck2D.cxx

```diff
@@ -0,0 +1,144 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   PadCalibrationCheck2D.cxx
+/// \author Laura Serksnyte
+///
+
+#include "TPC/PadCalibrationCheck2D.h"
+#include "QualityControl/MonitorObject.h"
+#include "QualityControl/Quality.h"
+
+#include <fairlogger/Logger.h>
+// ROOT
+#include <TCanvas.h>
+#include <TH2.h>
+#include <TPad.h>
+#include <TList.h>
+#include <TPaveText.h>
+
+using namespace std;
+
+namespace o2::quality_control_modules::tpc
+{
+
+void PadCalibrationCheck2D::configure(std::string) {}
+Quality PadCalibrationCheck2D::check(std::map<std::string, std::shared_ptr<MonitorObject>>* moMap)
+{
+
+  auto mo = moMap->begin()->second;
+  Quality result = Quality::Null;
+  if (mo->getName() == "c_ROCs_Pedestal_2D") {
+    result = Quality::Good;
+    auto* canv = dynamic_cast<TCanvas*>(mo->getObject());
+    // Check all histograms in the canvas
+    for (int tpads = 1; tpads <= 72; tpads++) {
+      auto padName = fmt::format("c_ROCs_Pedestal_2D_{:d}", tpads);
+      auto histName = fmt::format("h_Pedestals_ROC_{:02d}", tpads - 1);
+      TPad* pad = (TPad*)canv->GetListOfPrimitives()->FindObject(padName.data());
+      TH2F* h = (TH2F*)pad->GetListOfPrimitives()->FindObject(histName.data());
+      std::string titleh = h->GetTitle();
+      bool iroc = false;
+      bool oroc = false;
+      //check if we are dealing with IROC or OROC
+      if (titleh.find("IROC") != std::string::npos)
+        iroc = true;
+      if (titleh.find("OROC") != std::string::npos)
+        oroc = true;
+      if (!(iroc || oroc))
+        return Quality::Null;
```

Comment:
add readability braces and simplify expression (assuming it cannot be both, IROC and OROC). Also, the IROC / OROC distinction is only used to get the total number of pads.

```suggestion
      int totalPads = 0;
      if (titleh.find("IROC") != std::string::npos) {
        totalPads = 5280;
      } else if (titleh.find("OROC") != std::string::npos) {
        totalPads = 9280;
      } else {
        return Quality::Null;
      }
```